app/controllers/internal/badges_controller.rb

```diff
@@ -1,22 +1,29 @@
 class Internal::BadgesController < Internal::ApplicationController
   layout "internal"
+  before_action :ensure_badge, only: :award_badges
 
   def index
     @badges = Badge.all
   end
 
   def award_badges
     usernames = permitted_params[:usernames].split(/\s*,\s*/)
-    badge_slug = permitted_params[:badge]
     message = permitted_params[:message_markdown].presence || "Congrats!"
-    BadgeRewarder.award_badges(usernames, badge_slug, message)
+    BadgeRewarder.award_badges(usernames, permitted_params[:badge], message)
     flash[:success] = "BadgeRewarder task ran!"
     redirect_to internal_badges_url
+  rescue ArgumentError => e
+    flash[:danger] = e
```

Comment:
I ran into a serialization error and my guess is that this (and probably the [other internal controller that uses this 🙈](
https://github.com/thepracticaldev/dev.to/blob/59f5cd7e07251a35b9ba54c8a500bb69b1022762/app/controllers/internal/tools_controller.rb#L19)) should be `e.message`.

Reply:
Woops, good catch!